### DISCUSSION TITLE
Keymap suggestions: `F5`, `F10` etc. for dap and `gc` to comment a line

### DISCUSSION THREAD

**Discussion by qazxcdswe123** [LazyVim/LazyVim#637] [Ideas]
### Dap
Currently, to access dap command you need `<space>d?`, which is annoying. Here is a example config to use `F5`, `F10` etc. to setup dap, breakpoint, step into or more

```lua
{
    "mfussenegger/nvim-dap",
    keys = {
      { "<F5>", "<leader>dc", desc = "Continue", remap = true },
      { "<F17>", "<leader>dt", desc = "Terminate", remap = true }, -- Shift+F5
      -- stylua: ignore
      { "<F29>", function() require("dap").restart_frame() end, desc = "Restart frame", remap = true }, -- Ctrl+F5
      { "<F9>", "<leader>db", desc = "Toggle breakpoint", remap = true },
      { "<F10>", "<leader>dO", desc = "Step over", remap = true },
      { "<F11>", "<leader>di", desc = "Step into", remap = true },
      { "<F23>", "<leader>do", desc = "Step out", remap = true }, -- shift+F11
    },
  }
```

### `mini.comment`
To comment multiple line in visual mode, the keybind is `gc`, but to comment a line in normal mode, need `gcc`. This inconsistency is annoying and can be configured as

```lua
{
"echasnovski/mini.comment",
  opts = {
    mappings = {
      comment = "", --disable
      comment_line = "gc",
    },
  },
}
```

**Comment by Jomik** [comment]
Unsure if this is a suggestion for something to be changed in LazyVim, or just for someone to copy paste.

> ### Dap
> Currently, to access dap command you need `<space>d?`, which is annoying. Here is a example config to use `F5`, `F10` etc. to setup dap, breakpoint, step into or more

I really do not like using function keys, don't think I have ever really seen them bound in vim tbh 😄 
The `<leader>d` group fits perfectly from my point of view, and it is easy to add your own mappings.

> ### `mini.comment`
> To comment multiple line in visual mode, the keybind is `gc`, but to comment a line in normal mode, need `gcc`. This inconsistency is annoying and can be configured as

`gcc` is only to comment current line. Remember that you could also comment out other things. `gcio`, `gcii` and `gcif` comes to mind.

**Reply by qazxcdswe123** [reply]
> Unsure if this is a suggestion for something to be changed in LazyVim, or just for someone to copy paste.



Both :D



the comment part is mostly opinion based, I changed it mostly because of the inconsistency, haven't tried your suggestions, TIL, thanks.



But for the dap part, just out of curiosity, wouldn't it be bothersome to have 3 keystrokes just to move debug cursor, or is there some debugging technique I should learn? 

for me the most obvious way is to abuse `run to cursor` to avoid pressing `<leader>d?` multiple times

**Reply by Jomik** [reply]
Hmm, doesn't bother me that much - I rarely have that many breakpoints I would hit. I suppose step in and step out could be a bit bothersome? Haven't paid much attention to that yet though 🤔 
`run_to_cursor` isn't supported by all adapters, sadly.
